### ISSUE TITLE
use of unitialized value

### ISSUE THREAD

**Issue by nbecker** [serge-sans-paille/pythran#1229]

**Comment by nbecker** [comment]
This code causes memory corruption

```
import numpy as np

#pythran export empirical(float[:], float)
def empirical(ds, alpha):
    ds = np.sort(ds)
    ds_to_the_alpha = ds**alpha 
    fractions = ds_to_the_alpha / sum (ds_to_the_alpha)
    thresholds = np.cumsum(fractions)
    x = np.random.ranf()
    #i = find_first (self.thresholds, lambda u: x < u)
    i = np.searchsorted (thresholds, x)
    #print ('fractions:', fractions, 'thresh:', thresholds)
    print (thresholds, x, i)
    #print ('x:', x, 't:', self.thresholds, 'i:', i)
    return i
```
valgrind says:
```
==9177== Conditional jump or move depends on uninitialised value(s)
==9177==    at 0x18C918B6: __lower_bound<double const*, double, __gnu_cxx::__ops::_Iter_less_val> (stl_algobase.h:959)
==9177==    by 0x18C918B6: lower_bound<double const*, double> (stl_algobase.h:993)
==9177==    by 0x18C918B6: searchsorted<double, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long int> > > (searchsorted.hpp:27)
==9177==    by 0x18C918B6: operator()<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long int> >&, double&> (searchsorted.hpp:27)
==9177==    by 0x18C918B6: __pythran__empirical_prob::empirical::type<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> >&, double&>::result_type __pythran__empirical_prob::empirical::operator()<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> >&, double&>((anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> >&, double&) const [clone .isra.271] (tmp34bhbe8_.cpp:70)
==9177==    by 0x18C93377: empirical0 (tmp34bhbe8_.cpp:82)
==9177==    by 0x18C93377: __pythran_wrap_empirical0 (tmp34bhbe8_.cpp:102)
==9177==    by 0x18C93377: operator() (tmp34bhbe8_.cpp:114)
==9177==    by 0x18C93377: handle_python_exception<__pythran_wrapall_empirical(PyObject*, PyObject*, PyObject*)::<lambda()> > (exception_handler.hpp:17)
==9177==    by 0x18C93377: __pythran_wrapall_empirical(_object*, _object*, _object*) (tmp34bhbe8_.cpp:111)
```
Now if I uncomment the 1st print statement, the code runs normally :(

```
[compiler]
defines=USE_GMP USE_XSIMD
undefs=
include_dirs=
libs=gmp gmpxx stdc++ 
library_dirs=
cflags=-std=c++11 -fno-math-errno -w -O3 -ffast-math -march=native
#cflags=-std=c++11 
ldflags=
CXX=g++
#CXX=clang++
CC=gcc
#CC=clang
blas=openblas
ignoreflags=-flto-partition=none -fstack-clash-protection
[pythran]
complex_hook=True
```
g++ (GCC) 8.2.1 20181215 (Red Hat 8.2.1-6)

**Comment by nbecker** [comment]
With the code above (with the single uncommented print) I get printed:
[0.235228        0        0        06.90753e-3106.90753e-310] 0.484901 6
which shows that "thresholds" has been corrupted

**Comment by serge-sans-paille** [comment]
@nbecker I don't fully understand why but #1230 fixes the issue on my side... can you confirm?

**Comment by serge-sans-paille** [comment]
@nbecker bug spotted and fixed (or so I think) by #1230 update

**Comment by nbecker** [comment]
#1230 seems to fix my issue, thanks!
